mono/utils/mono-time.c

```diff
@@ -161,6 +178,31 @@ mono_100ns_ticks (void)
 }
 
 /*
+ * Returns user + system time for a thread in 100ns units.
+ */
+gint64
+mono_thread_cpu_time (void)
+{
+#if defined(__linux__)
+	struct timespec ts;
+	clock_gettime (CLOCK_THREAD_CPUTIME_ID, &ts);
+	return ts.tv_sec * 10000000 + ts.tv_nsec / 100;
+#elif defined(PLATFORM_MACOSX)
+	mach_msg_type_number_t count = THREAD_BASIC_INFO_COUNT;
+	thread_basic_info_data_t info;
+
+	if (thread_info (pthread_mach_thread_np (pthread_self ()), THREAD_BASIC_INFO,
```

Comment:
3 issues here:

1)Don't use pthread_mach_thread_np.
The implementation takes a global lock and does a linear search over all live threads.
Use mach_thread_self instead.

2)Your code is leaking a port rights. Every time you fetch a mach port, its  ref count is increases and you need to decrement at the end.
You can use mach_port_deallocate to decrease the ref count.
You can use mach_port_get_refs to check the ref count of a specific port.

3)There's no reason on why this code would not work on iOS.
Use PLATFORM_DARWIN for this condition